RoofingCompany.sql

```diff
@@ -989,10 +982,106 @@ GO
 
 CREATE VIEW vHRExamination
 AS
-SELECT IdMedicalExamination
-FROM MedicalExamination
-WHERE  MedicalExamination.Date < DATEADD(day, 14, GETDATE());
+SELECT DISTINCT Employee.IdEmployee, Employee.EmployeeName, Employee.EmployeeSurname, Employee.PESEL, Position.Workplace, Position.VailidityOfMedicalExam, (ISNULL(MedicalExamination.Date, '2000-01-01')) as PreviousExamination, (ISNULL(DATEADD(day, VailidityOfMedicalExam, MedicalExamination.Date),CONVERT(date,GETDATE()))) as NextExamination
+FROM Employee 
+inner JOIN Contract ON Employee.IdEmployee = Contract.IdEmployee 
+inner JOIN Position ON Contract.IdPosition = Position.IdPosition 
+left outer JOIN MedicalExamination ON Employee.IdEmployee = MedicalExamination.IdEmployee
+Where (Contract.EndDate > GETDATE()) AND (DATEADD(Day, VailidityOfMedicalExam,(ISNULL(MedicalExamination.Date,'2000-01-01')))) < GETDATE()+14
+GO
+
+CREATE VIEW vAbsenceType
+AS
+SELECT IdAbsenceType, AbscenceReason
+FROM AbsenceType
+
+GO
 
+CREATE VIEW vExpencePartOrder
+AS
+SELECT Supplier.IdSupplier, Supplier.SupplierName, PartOrder.OrderDate, PartOrder.StatusPartOrder, PartOrder.CostPartOrder
+FROM PartOrder 
+INNER JOIN Supplier ON PartOrder.IdSupplier = Supplier.IdSupplier
+
+GO
+
+CREATE VIEW vEducationLevel
+AS
+SELECT IdEducationLevel, EducationLevel, Degree
+FROM EducationLevel
+
+GO
+
+CREATE VIEW vSalariesSummary
+as
+WITH 
+L4 AS (SELECT Employee.IdEmployee, SUM(Datediff(day, StartOfAbsence, EndOfAbsence))+1 as 'L4Days', AbsenceType.Multiplier as 'L4X'
```

Comment:
to wynieść do oddzielnego widoku

Reply:
@rudnickirobert wszystkie rodzaje nieobecności? bo to specjalnie zostało tak zrobione żeby w podsumowaniu do wypłaty były widoczne wszystkie nieobecne dni w poprzednim miesiącu bez Tworzenia dodatkowych widoków do tego

Reply:
Poprawione :) wszystkie powody nieobecności w oddzielnych widokach